python/vyos/ifconfig/utunnel.py

```diff
@@ -0,0 +1,122 @@
+# Copyright 2019-2021 VyOS maintainers and contributors <maintainers@vyos.io>
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library.  If not, see <http://www.gnu.org/licenses/>.
+import glob
+import os
+from pathlib import Path
+import yaml
+
+from vyos.ifconfig import Interface
+from vyos.ifconfig import Operational
+
+utunnel_config_directory = '/config/utunnels/'
+
+
+def get_utunnel_config(tunnel_type):
+    config_file = os.path.join(utunnel_config_directory, '{}.yaml'.format(tunnel_type))
```

Comment:
Instead of hardcoding some `/config/utunnels/.yml` why not generate .yml files from config/XML and do not use `/config/utunnels` at all?
Copy config on one server and paste on another is a convenient, but we will loose this from such hacks.